### ISSUE TITLE
Seg Fault in HydroDyn in v3.5.1

### ISSUE THREAD

**Issue by dzalkind** [OpenFAST/openfast#1863]
**Bug description**
I'm getting a seg fault when running OpenFAST v3.5.1 (via conda installation, both locally and on a linux CI) when running HydroDyn and our IEA-22MW reference model.

**To Reproduce**
<Update the following list with your specific information.>
1. `conda install openfast`
2. clone/download the I[EA-22MW reference model](https://github.com/IEAWindTask37/IEA-22-280-RWT/tree/main/OpenFAST/IEA-22-280-RWT-Monopile)
3. run `openfast IEA-22-280-RWT-Monopile.fst`
4. See the error

**Expected behavior**
A simple test simulation with most modules enabled.

**Screenshots, if applicable**
```
 **************************************************************************************************
 OpenFAST

 Copyright (C) 2023 National Renewable Energy Laboratory
 Copyright (C) 2023 Envision Energy USA LTD

 This program is licensed under Apache License Version 2.0 and comes with ABSOLUTELY NO WARRANTY.
 See the "LICENSE" file distributed with this software for details.
 **************************************************************************************************

 OpenFAST-HEAD-HASH-NOTFOUND
 Compile Info:
  - Compiler: GCC version 12.3.0
  - Architecture: 64 bit
  - Precision: single
  - OpenMP: Yes, number of threads: 16/16
  - Date: Oct 20 2023
  - Time: 23:54:21
 Execution Info:
  - Date: 11/08/2023
  - Time: 13:33:58-0700

 OpenFAST input file heading:
     Generated with AeroElasticSE FAST driver

 Running ElastoDyn.
 Nodal outputs section of ElastoDyn input file not found or improperly formatted.
 Running AeroDyn.
 AD15 Nodal Outputs: Nodal output section of AeroDyn input file not found or improperly formatted.
 Skipping nodal outputs.
 Running InflowWind.
 Running HydroDyn.

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x10aeaad96
#1  0x10aea9edd
#2  0x7ff8101a35ec
#3  0x109645540
#4  0x10961af00
#5  0x1088163c3
#6  0x10881fd76
#7  0x1086d6a65
#8  0x109e95b5e
#9  0x7ff80fe1c41e
```

**OpenFAST Version**
```
(iea-env) dzalkind@x86_64-apple-darwin13 IEA-22-280-RWT % conda list | grep openfast
openfast                  3.5.1                h157c8df_0    conda-forge
```

**Additional context**
Workarounds (clues :-)):
- Setting `CompHydro` to 0 runs okay.  
- Adding `Wave1Elev` to the output list also works fine.
- OpenFAST v3.5.0 works okay, too.

**Comment by andrew-platt** [comment]
@luwang00, do you have any ideas?

**Comment by bjonkman** [comment]
@dzalkind, does this happen only when HydroDyn doesn't contain ANY outputs? or just when it doesn't contain that specific `Wave1Elev` output?

I recently found an error in the dev branch (which is significantly different than v3.5.1) that could attempt to pass an unallocated array when no output was selected.

**Comment by dzalkind** [comment]
It appears as though it wants _some_ output specified.  Does not have to be `Wave1Elev`.

**Comment by luwang00** [comment]
I also vaguely remember there was a bug with HydroDyn when there's no output channel requested, but I can't find that particular issue.

**Comment by andrew-platt** [comment]
I remember seeing this recently as well.  Not sure where though. @bjonkman, I suspect the unallocated array passing you found is the issue.

**Comment by deslaughter** [comment]
The issue is around Line 3081 of `HydroDyn_Input.f90` from the `3.5.1` release. If no outputs are specified then `InputFileData%Waves2%NumOuts`, `InputFileData%Morison%NumOuts`, and `InputFileData%NumOuts` are not set and `InputFileData%OutList` is not allocated. @bjonkman fixed this in PR #1008 (https://github.com/OpenFAST/openfast/pull/1008/commits/c9a6153e336ebe50ff207a28284f8d634c9a56e3). @andrew-platt should we fix this in a 3.5.2 release?

**Comment by deslaughter** [comment]
The `rc-3.5.2` branch was created to address this issue and other potential issues with `v3.5`. You can either use the new branch or add any output variable to HydroDyn.
